server/src/main/kotlin/net/horizonsend/ion/server/generation/PersistentDataTypes.kt

```diff
@@ -0,0 +1,75 @@
+package net.horizonsend.ion.server.generation
+
+import net.horizonsend.ion.server.generation.configuration.Palette
+import org.apache.commons.lang.SerializationUtils
+import org.bukkit.persistence.PersistentDataAdapterContext
+import org.bukkit.persistence.PersistentDataType
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.io.ObjectInputStream
+import java.io.Serializable
+
+data class Asteroid(
+	val x: Int,
+	val y: Int,
+	val z: Int,
+	val palette: Palette,
+	val size: Double,
+	val octaves: Int
+) : Serializable
+
+data class Asteroids(val asteroids: List<Asteroid>) : Serializable
+
+class AsteroidsDataType : PersistentDataType<ByteArray, Asteroids> {
+	override fun getPrimitiveType(): Class<ByteArray> {
+		return ByteArray::class.java
+	}
+
+	override fun getComplexType(): Class<Asteroids> {
+		return Asteroids::class.java
+	}
+
+	override fun fromPrimitive(primitive: ByteArray, context: PersistentDataAdapterContext): Asteroids {
+		val inputStream: InputStream = ByteArrayInputStream(primitive)
```

Comment:
no. you should construct an object properly

Reply:
pls explain